### PR TITLE
Feature process documentation and contribution update

### PR DESCRIPTION
**Pull request by StephenHodgson**
Overview
---
Updated contribution guidelines and added feature area process documentation.

### CODE REVIEW

File: Feature_Contribution_Process.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+* Be registered with the Mixed Reality Manager.
+    * If the feature is a core feature, this should be hard coded into the `MixedRealityManager` and added to the `MixedRealityConfigurationProfile`.
+        * This includes being able to specify a concrete implementation via dropdown using `SystemType`.
+        * A default configuration profile located in `MixedRealityToolkit-SDK/Profiles` and be assigned in the default configuration profile for the Mixed Reality Manager
+    * If this feature is **not** a core feature, then it must be registered using the component configuration profile and implement `IMixedRealityComponent`.
+* Have a default implementation located in `MixedRealityToolkit-SDK/Features/<FeatureName>`
+* Events that can be raised with the system should be defined in the interface, with all the required parameters for initializing the event data.
+
+## Event Data Implementation Requirements
+The EventData defines exactly what data the handler is expected to receive from the event.
```
Comment (StephenHodgson) [inline Feature_Contribution_Process.md]: Event Data

File: Feature_Contribution_Process.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+* All Event Datum for the feature should be defined in `MixedRealityToolkit/_Core/EventDatum/<FeatureName>`.
+* All new Event Data classes should inherit from `GenericBaseEventData`
+
+## Handler Implementation Requirements
+
+The HandLer Interface defines each event a component should be listening for and the types of data passed. End users will implement the interface to execute logic based on the event data received.
```
Comment (StephenHodgson) [inline Feature_Contribution_Process.md]: HandLer -> Handler

**Comment by StephenHodgson** [comment]
@mr0ng

**Comment by SimonDarksideJ** [comment]
@ryzngard any chance you can do a quick PR agsinst @StephenHodgson fork like I did to push this through quicker.  Thanks for spotting those.

**Comment by StephenHodgson** [comment]
Have you guys watched this video about dependency validation?

I'm wondering if we can add these rules and ppl that download the mrtk will still get these kinds of errors, or warnings.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,70 +1,105 @@
...
+To ensure a smooth process when contributing new fixes or features, it's key that you start your journey by creating one of the issue types listed above.
+
+Start by [opening a proposal](https://github.com/Microsoft/MixedRealityToolkit-Unity/issues/new/choose) describing the change you want to make and how your proposed implementation, or simply the issue you are facing. This will enable us have a brief discussion about the proposal and, hopefully, identify some potential pitfalls before any work is started.
+
+If you're proposing a completely new feature (or a new platform support) please follow the [Feature Contribution Process](./Feature_Contribution_Process.md).
+
+>Note:  If you wish to work on something that already exists on our backlog, you can use that work item as your proposal. Be sure to also comment on the task notifying maintainers that you're working towards completing it.
+
+# Beginning development
+Working with Git, the contribution process is quite simply (provided you have installed a good Git Client such as TortoiseGit or SourceTree)
```
Comment (davidkline-ms) [inline CONTRIBUTING.md]: simply -> simple

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,70 +1,105 @@
...
+
+The Mixed Reality Toolkit requires the following forms of documentation for any new feature or component.  Also ensure if you are simply patching / fixing an existing feature that the documentation is also updated to match.
+
+### APIs
+
+This documentation is generated from the product code and is reviewed as part of **all** pull requests.
+
+### Conceptual
+
+Conceptual documentation is hosted on https://docs.microsoft.com/en-us/windows/mixed-reality. Please submit our changes via Pull Request at https://github.com/MicrosoftDocs/mixed-reality.
```
Comment (davidkline-ms) [inline CONTRIBUTING.md]: our -> your

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,70 +1,105 @@
...
+
+As part of your pull request, please update (or create) the Readme markdown file in the appropriate feature folder. This will allow GitHub users to gain a high-level understanding of your new feature.
+
+## Supported Platforms
+
+The Mixed Reality Toolkit supports the following mixed reality (AR/VR/XR) platforms:
+
+- Windows Standalone
+    - OpenVR
+- Windows Mixed Reality
```
Comment (davidkline-ms) [inline CONTRIBUTING.md]: Universal Windows Platform

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,70 +1,105 @@
...
+    - Immersive devices
```
Comment (davidkline-ms) [inline CONTRIBUTING.md]: Windows Mixed Reality Immersive devices

File: Feature_Contribution_Process.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
+# Feature Contribution Process
+
+Adding features to the Mixed Reality Toolkit (MRTK) is split up into a few iteration steps, so maintainers can have time to review and and ensure the process goes smoothly. Please be sure to review the list of [feature requirements](#new-feature-requirements) before you get started.
+
+# Process
+The following process has been drafted to ensure all new work complies to the updated standards and architecture defined for the MRTK, this has been defined as:
+
+1. [Open a new Proposal and related Tasks](#new-proposal)
+2. [Submit an Architecture Draft or Outline](#architecture-draft)
+3. [Review and finalization Architecture documentation](#architecture-documentation)
```
Comment (davidkline-ms) [inline Feature_Contribution_Process.md]: finalization -> finalize the

File: Feature_Contribution_Process.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+## Core Implementation
+
+The initial work that should be submitted, is to implement:
+
+* Definitions
+* Interfaces
+* Configuration profiles
+* Event data
+
+And other components required by the the Mixed Reality Toolkit's core framework.
```
Comment (davidkline-ms) [inline Feature_Contribution_Process.md]: not sure I get what you mean by this line?

**Comment by StephenHodgson** [comment]
@davidkline-ms please re-review :)

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -100,6 +100,7 @@ The Mixed Reality Toolkit supports the following mixed reality (AR/VR/XR) platfo
 
 - Windows Standalone
     - OpenVR
-- Windows Mixed Reality
-    - Immersive devices
-    - Microsoft HoloLens
\ No newline at end of file
+- Universal Windows Platform
+    - Standalone PC
```
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply CONTRIBUTING.md]: Yes. and we already do, so long as we can get mouse support back on the config profile. (pending bug fixes)

[the remark this answers is not in the document]
Comment (Ecnassianer) [reply CONTRIBUTING.md]: Super useful for debugging! :)

Also, it's a good reminder that our project may end up in headless servers, asymmetric multiplayer clients and other non-MR experiences, which requires at a minimum that it doesn't fail to build for Standalone PC.

**Pull request merged**